### ISSUE TITLE
using parameters with wildcard for string matching doesn't work

### ISSUE THREAD

**Issue by Balnian** [bgrainger/MySqlConnector#32]
This is the minimum to reproduce the bug  : 
```C#
try
{
    using (var connection = new MySqlConnection(<ConnectionString>))
    {
        connection.Open();
        using (var cmd = connection.CreateCommand())
        {
            cmd.CommandText = @"SELECT <ColumnName>FROM <Table> WHERE <ColumnName>LIKE ':id';";
            var cmdParam = cmd.CreateParameter();
            cmdParam.DbType = DbType.String;
            cmdParam.ParameterName = ":id";
            cmdParam.Value = $@"%{<StringToMatch>}%";
            cmd.Parameters.Add(cmdParam);
            using (var reader = await cmd.ExecuteReaderAsync().ConfigureAwait(false))
            {
                while (reader.Read())
                {
                    Console.WriteLine(reader[<ColumnName>]);
                }
            }
        }
    }

}
catch (Exception ex)
{
    Console.WriteLine(ex.Message);
}
```
In the results view of the reader 
		Message:	"Specified method is not supported."	string
		Source:	"MySqlConnector"	string
		StackTrace:	"   at MySql.Data.MySqlClient.MySqlDataReader.GetEnumerator()\r\n   at System.Linq.SystemCore_EnumerableDebugView.get_Items()"	string

**Comment by bgrainger** [comment]
This appears to be caused by `System.Linq.SystemCore_EnumerableDebugView.get_Items()`.

The underlying issue is that `DbDataReader.GetEnumerator` is not implemented #33 but this code snippet doesn't show any calls to that method; is it perhaps caused by some kind of debugger visualisation or watch window?

Additionally, I'm not familiar with parameter interpolation being performed *inside* a string literal, i.e., treating `:id` as a symbol in the literal `LIKE ':id'`. Do you have a link to documentation describing this? The [MySQL documentation](https://dev.mysql.com/doc/connector-net/en/connector-net-tutorials-parameters.html) only describes using `@` to designate parameter names, not `:`.

**Comment by bgrainger** [comment]
I've shipped [0.1.0-alpha13](https://www.nuget.org/packages/MySqlConnector/0.1.0-alpha13) with a fix for the `NotImplementedException` in `DbDataReader.GetEnumerator`.

Can you still reproduce the bug with this latest version?

**Comment by Balnian** [comment]
It's fixed thanks a lot!!!
